src/custom/utils/paraswap.ts

```diff
@@ -0,0 +1,86 @@
+import { OrderKind } from '@gnosis.pm/gp-v2-contracts'
+import { ParaSwap, SwapSide, NetworkID } from 'paraswap'
+import { toErc20Address } from 'utils/tokens'
+import { PriceQuoteParams } from 'utils/operator'
+import { OptimalRatesWithPartnerFees, RateOptions } from 'paraswap/build/types'
+import { ChainId } from '@uniswap/sdk'
+import { PriceInformation } from 'state/price/reducer'
+import { getTokensFromMarket } from './misc'
+
+type ParaSwapPriceQuote = OptimalRatesWithPartnerFees
+
+// Provided manually just to make sure it matches what GPv2 backend is using, although the value used  is the current SDK default
+const API_URL = 'https://apiv4.paraswap.io/v2'
+
+const parSwapLibs: Map<ChainId, ParaSwap> = new Map()
+
+function getParaswapChainId(chainId: ChainId): NetworkID | null {
+  switch (chainId) {
+    // case ChainId.RINKEBY: // Rinkeby is supported by the SDK but not from the API
```

Comment:
I get that they only accept mainnet but maybe just comment that directly

Reply:
I did some changes, and provided a link to the API so is easier to add if they have xDAI or Rinkeby in the future